app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +280,143 @@ angular.module('confRegistrationWebApp')
     $scope.hasPendingCheckPayment = function(payments){
       return _.some(payments, { paymentType: 'CHECK', status: 'PENDING' });
     };
+
+    ////// START EVENT-433 //////
+
+    // Get spouse registration
+    // End point: getSpouseRegistration = /conferences/conferenceId/registrations/spouse
+    ///// TEST DATA change end point url below to: /conferences/conferenceId/registrations/current /////
+    $http.get('conferences/' + conference.id + '/registrations/spouse')
+      .then(function(response) {
+        if (response.status === 200) {
+          $scope.spouseRegistration = response.data;
+
+          ///// TEST DATA /////
+          // $scope.spouseRegistration.completed = true;
+          // $scope.spouseRegistration.registrants[0].email = 'janeworner3@hotmail.com';
+          // $scope.spouseRegistration.registrants[0].firstName = 'Jane';
+          // $scope.spouseRegistration.registrants[0].lastName = 'Worner3';
+          ///// TEST DATA ////
+
+        }
+        else {
+          $scope.spouseRegistration = {};
+          console.log('No current and/or spouse registration found for user with ID.');
+        }
+      }).catch(function(response) {
+      $scope.spouseRegistration = {};
+      console.log('No current and/or spouse registration found for user with ID. Status = ' + response.status);
+      console.log(response.data);
+    })
+      .finally(function() {
+        checkSpouseRegistration();
+      });
+
+    //Parameters needed to retrieve all current registrants for conference.id
+    var queryParameters = {
+      block: [],
+      //page: 1,
+      //limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes'
+    };
+
+    //Get all current registrants based on conference.id and queryParameters
+    //then check if spouse is registered and display 'Spouse Registration' section if needed.
+    var checkSpouseRegistration = function() { // jshint ignore:line
+      $http.get('conferences/' + conference.id + '/registrations', {params: queryParameters})
+        .then(function (response) {
+          var _spouseSectionVisible = sessionStorage.getItem('spouseSectionVisible');
+          //var spouseIncludedOnCurrentRegistration = _.contains(_.map($scope.spouseRegistration.registrants, 'email'), $scope.currentRegistration.registrants[0].email);
+          var spouseIncludedOnCurrentRegistration = !_.isEmpty(_.intersection(_.map($scope.spouseRegistration.registrants, 'email'),
+            _.map(_.map(_.map(response.data.registrations, 'registrants'), 0), 'email')));
+
+          // 2. If the registrant does have a spouse who has already registered, then
+          if ($scope.spouseRegistration.completed) {
```

Comment:
After a discussion with Lee, we have concluded that we should consider spouse registrations that are incomplete instead of ignoring them like we are doing with this condition. So instead of checking the `completed` property, we only need to check if the spouse registration exists. The way that I would do that is to set `$scope.spouseRegistration` to `null` instead of `{}` if the API returns a 404 error and then here check if `$scope.spouseRegistration !== null`.